src/main/java/bio/terra/service/resourcemanagement/flight/DeleteLogAnalyticsWorkspaceStep.java

```diff
@@ -0,0 +1,50 @@
+package bio.terra.service.resourcemanagement.flight;
+
+import bio.terra.model.BillingProfileModel;
+import bio.terra.service.job.DefaultUndoStep;
+import bio.terra.service.profile.flight.ProfileMapKeys;
+import bio.terra.service.resourcemanagement.azure.AzureMonitoringService;
+import bio.terra.service.resourcemanagement.azure.AzureStorageAccountResource;
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.FlightMap;
+import bio.terra.stairway.StepResult;
+import bio.terra.stairway.exception.RetryException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import java.util.List;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class DeleteLogAnalyticsWorkspaceStep extends DefaultUndoStep {
+  private static final Logger logger =
+      LoggerFactory.getLogger(DeleteLogAnalyticsWorkspaceStep.class);
+  private AzureMonitoringService monitoringService;
```

Comment:
field can be `final`
```suggestion
  private final AzureMonitoringService monitoringService;
```